### PR TITLE
`resource exists` command for existence check of arm resources.

### PR DESCRIPTION
**Pull request by williexu**
---
Will address the needs of some users who want to check the existence of a resource before creating.
Might address: https://github.com/Azure/azure-cli/issues/6442
First step towards resolving: https://github.com/Azure/azure-cli/issues/5006

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by troydai** [comment]
is this for this sprint?

**Comment by troydai** [comment]
I suggest you explicitly state in the help doc that the exit code indicate if the resource exists rather than if the command successes.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -693,7 +693,26 @@ def _single_or_collection(obj, default=None):
...
+                                                                              resource_name)]
+    from azure.cli.core.util import empty_on_404
+
+    def _check_existence(id_dict):
+        try:
+            _get_rsrc_util_from_parsed_id(cmd.cli_ctx, id_dict, api_version).get_resource()
+        except Exception as ex:  # pylint: disable=broad-except
+            empty_on_404(ex)
+            return False
+        return True
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: Should this simply return True/False or a JSON object like `{'exists': 'True/False'}`? Check with @johanste

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 2]
```diff
@@ -324,6 +324,27 @@
             az resource show -g MyResourceGroup --namespace Microsoft.Network --parent applicationGateways/ag1/urlPathMaps/map1 --resource-type pathRules -n rule1
 """
 
+helps['resource exists'] = """
+    type: command
+    short-summary: Check the existence of a resource.
+    examples:
+        - name: Check existence of a virtual machine resource named 'MyVm'.
+          text: >
+            az vm exists -g MyResourceGroup -n MyVm --resource-type "Microsoft.Compute/virtualMachines"
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: should be `resource exists`

**Comment by williexu** [comment]
@troydai NOT for this sprint

**Comment by williexu** [comment]
Closing this as https://github.com/Azure/azure-cli/pull/6489 will allow for an existence check.

**Pull request closed without merging**